jupyter-extension/jupyterlab_pachyderm/handlers.py

```diff
@@ -274,17 +275,32 @@ class PPSCreateHandler(BaseHandler):
 
     @tornado.web.authenticated
     async def get(self, path):
-        """Get the pipeline spec for the specified notebook."""
-        response = await self.pps_client.generate(path)
-        await self.finish(response)
+        try:
+            """Get the pipeline spec for the specified notebook."""
+            response = await self.pps_client.generate(path)
+            get_logger().debug(f"GetPipelineSpec: {response}")
+            await self.finish(response)
+        except Exception as e:
+            if isinstance(e, tornado.web.HTTPError):
+                # Common case: only way to print the "reason" field of HTTPError
+                get_logger().error(f"couldn't create pipeline: {e.reason}")
+            get_logger().error("\n".join(traceback.format_exception(e)))
```

Comment:
Is this supposed to be in an else block?

Reply:
I deliberately put it outside the `if`, so that we'd log the trace no matter what (while I was figuring out how to pass errors through to the UI, that was my best tool for understanding what was going on). Happy to move it to an `else` or format it differently or something?